client/src/app/pages/applications/components/application-detail-drawer/application-detail-drawer-analysis.tsx

```diff
@@ -134,25 +131,50 @@ export const ApplicationDetailDrawerAnalysis: React.FC<
                 >
                   Analysis details
                 </Button>
-                <SimpleDocumentViewerModal<Task | string>
-                  title={`Analysis details for ${application?.name}`}
-                  fetch={getTaskById}
-                  documentId={taskIdToView}
-                  onClose={() => setTaskIdToView(undefined)}
-                />
               </>
             ) : (
               <span className={spacing.mlSm}>
-                <ExclamationCircleIcon color="#c9190b"></ExclamationCircleIcon>
+                <ExclamationCircleIcon
+                  color={COLOR_NAMES_BY_HEX_VALUE.red}
+                ></ExclamationCircleIcon>
                 Failed
               </span>
             )
           ) : (
-            notAvailable
+            <>
+              {task ? (
+                <Button
+                  icon={
+                    <span className={spacing.mrXs}>
+                      <ExclamationCircleIcon
+                        color={COLOR_NAMES_BY_HEX_VALUE.blue}
+                      ></ExclamationCircleIcon>
+                    </span>
+                  }
+                  type="button"
+                  variant="link"
+                  onClick={() => setTaskIdToView(task?.id)}
+                  className={spacing.ml_0}
+                  style={{ margin: "0", padding: "0" }}
+                >
+                  Analysis details
+                </Button>
+              ) : null}
+
+              {notAvailable}
```

Comment:
This `notAvailable` should be on the else end of the `task ? <Button> : null` ternary instead of null, right? Currently you get both the "Analysis details" button and then "not available" text below it.

Reply:
Yeah - I was wondering what we should show in this case for the "catch all status". I think your suggestion makes sense.